protocol/genesis/types/share.go

```diff
@@ -15,53 +15,44 @@ func ConvertToGenesisShare(share *spectypes.Share, operator *spectypes.Committee
 
 	genesisShare := &genesisspectypes.Share{
 		OperatorID:          operator.OperatorID,
-		SharePubKey:         share.SharePubKey,
+		SharePubKey:         make([]byte, len(share.SharePubKey)),
 		ValidatorPubKey:     key,
 		Committee:           make([]*genesisspectypes.Operator, 0, len(share.Committee)),
 		Quorum:              q,
 		PartialQuorum:       pc,
 		DomainType:          genesisspectypes.DomainType(share.DomainType),
 		FeeRecipientAddress: share.FeeRecipientAddress,
-		Graffiti:            share.Graffiti,
+		Graffiti:            make([]byte, len(share.Graffiti)),
 	}
 
+	copy(genesisShare.SharePubKey, share.SharePubKey)
+	copy(genesisShare.Graffiti, share.Graffiti)
+
 	for _, c := range share.Committee {
-		genesisShare.Committee = append(genesisShare.Committee, &genesisspectypes.Operator{
+		newMember := &genesisspectypes.Operator{
 			OperatorID: c.Signer,
-			PubKey:     c.SharePubKey,
-		})
+			PubKey:     make([]byte, len(c.SharePubKey)),
+		}
+		copy(newMember.PubKey, c.SharePubKey)
+		genesisShare.Committee = append(genesisShare.Committee, newMember)
 	}
 
 	return genesisShare
 }
 
-// TODO: (Alan) write tests with all fields and equality checks
-func ConvertToAlanShare(alanShare *typesv2.SSVShare, operator *spectypes.CommitteeMember) *SSVShare {
-	q, pc := ComputeQuorumAndPartialQuorum(len(alanShare.Committee))
-
-	key := make([]byte, len(alanShare.ValidatorPubKey))
-	copy(key, alanShare.ValidatorPubKey[:])
+// ConvertToGenesisSSVShare converts an Alan share to a genesis SSV share.
+func ConvertToGenesisSSVShare(alanSSVShare *typesv2.SSVShare, operator *spectypes.CommitteeMember) *SSVShare {
+	genesisShare := ConvertToGenesisShare(&alanSSVShare.Share, operator)
 
-	share := &SSVShare{
-		Share: genesisspectypes.Share{
-			OperatorID:          operator.OperatorID,
-			ValidatorPubKey:     key, // Ensure this is necessary; remove if ValidatorPubKey is already a slice.
-			SharePubKey:         alanShare.SharePubKey,
-			Committee:           make([]*genesisspectypes.Operator, 0, len(alanShare.Committee)),
-			Quorum:              q,
-			PartialQuorum:       pc,
-			DomainType:          genesisspectypes.DomainType(alanShare.DomainType),
-			FeeRecipientAddress: alanShare.FeeRecipientAddress,
-			Graffiti:            alanShare.Graffiti,
-		},
+	convertedMetadata := Metadata{
+		BeaconMetadata: alanSSVShare.Metadata.BeaconMetadata,
+		OwnerAddress:   alanSSVShare.Metadata.OwnerAddress,
+		Liquidated:     alanSSVShare.Metadata.Liquidated,
+		// lastUpdated field is not converted because it's unexported
 	}
 
-	for _, c := range alanShare.Committee {
-		share.Committee = append(share.Committee, &genesisspectypes.Operator{
-			OperatorID: c.Signer,
-			PubKey:     c.SharePubKey,
-		})
+	return &SSVShare{
+		Share:    *genesisShare,
+		Metadata: convertedMetadata,
 	}
```

Comment:
I missed it before, but we're only testing the spectypes.Share conversion and not the entire SSVShare, so this part remains untested.

With that said, I noticed that `Validator`(for whom we convert the SSVShare) only uses the Share.Share field and not the Metadata, so that could be unnecessary to test for.

@y0sher wdyt?

Reply:
My initial intention was to include metadata conversion for completeness. However, since the Validator isn't currently using it and the genesis code will be removed after the fork, it makes sense to remove the metadata conversion to simplify the code. wdyt?

Reply:
I think it's safe to leave it as it's not used in practice even though Validator struct still requires it.